pkg/tasks/tasks.go

```diff
@@ -110,7 +115,21 @@ type TaskSpec struct {
 type Task interface {
 	Run(ctx context.Context) error
 }
-type taskErr struct {
-	error
-	name string
+
+type TaskErr struct {
+	Err  error
+	Name string
+}
+
+type TaskGroupErrors []TaskErr
+
+func (tge TaskGroupErrors) Error() string {
+	if len(tge) == 0 {
```

Comment:
```suggestion
	if tge == nil {
```

Reply:
It's best to check the length here since an empty slice may not be nil, but will always have a length of 0.